src/main/java/com/puppycrawl/tools/checkstyle/PackageNamesLoader.java

```diff
@@ -159,7 +160,7 @@ public void endElement(String uri,
             throw new CheckstyleException("unable to open one of package files", ex);
         }
 
-        return result;
+        return Collections.unmodifiableSet(result);
```

Comment:
This one is a true-positve. Fixed.

Reply:
it is static function, collection is created internal as local variable.
Are you sure that it should be fixed ?

Reply:
You are right, this is a false positive. `result` is a local variable, no internal state is exposed here. Reverted.
```